npm_and_yarn/lib/dependabot/npm_and_yarn/helpers.rb

```diff
@@ -24,12 +57,39 @@ def self.npm_version_numeric(lockfile)
         6
       end
 
+      # Determines the npm version based on the lockfile version
+      # - NPM 7 uses lockfileVersion 2
+      # - NPM 8 uses lockfileVersion 2
+      # - NPM 9 uses lockfileVersion 3
+      sig { params(lockfile: DependencyFile).returns(Integer) }
+      def self.npm_version_numeric_npm8_or_higher(lockfile)
+        lockfile_content = lockfile.content
+
+        # Return default NPM version if there's no lockfile or it's empty
+        return NPM_DEFAULT_VERSION if lockfile_content.nil? || lockfile_content.strip.empty?
+
+        parsed_lockfile = JSON.parse(lockfile_content)
+
+        lockfile_version_str = parsed_lockfile["lockfileVersion"]
+
+        # Default to npm default version if lockfileVersion is missing or empty
+        return NPM_DEFAULT_VERSION if lockfile_version_str.nil? || lockfile_version_str.to_s.strip.empty?
+
+        lockfile_version = lockfile_version_str.to_i
+
+        return NPM_V9 if lockfile_version == 3
+
+        NPM_DEFAULT_VERSION
+      rescue JSON::ParserError
+        NPM_FALLBACK_VERSION # Fallback to default npm version if parsing fails
```

Comment:
any reason why default won't be the same as fallback?

Reply:
I removed fallback version and using default version always.